packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -58,10 +58,11 @@ export const TooltipPlugin: React.FC<TooltipPluginProps> = ({ mode = 'single', t
 
         // when interacting with a point in single mode
         if (mode === 'single' && originFieldIndex !== null) {
-          const field = otherProps.data[originFieldIndex.frameIndex].fields[originFieldIndex.fieldIndex];
+          const field = graphContext.alignedData.fields[focusedSeriesIdx!];
```

Comment:
@leeoniya having the aligned data coming into the panel would actually help reducin this context consumption. We could rely on the single data prop rather than distinguishing between aligned and not aligned data.